cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/HttpHandlerGenerator.java

```diff
@@ -380,10 +386,23 @@ public void visitEnd() {
       // If any annotations of the method is one of those HttpMethod,
       // this is a handler process, hence need to copy.
       boolean handlerMethod = false;
+      TransactionControl txCtrl = TransactionControl.IMPLICIT;
       for (AnnotationNode annotation : annotations) {
         if (isHandlerMethod(Type.getType(annotation.desc))) {
           handlerMethod = true;
-          break;
+        } else if (TX_POLICY_TYPE.equals(Type.getType(annotation.desc))) {
+          List list = annotation.values;
+          for (Iterator iter = list.iterator(); iter.hasNext(); ) {
+            if ("value".equals(iter.next()) && iter.hasNext()) {
+              Object value = iter.next();
+              if (String[].class.equals(value.getClass())) {
+                String[] values = (String[]) value;
+                if (values.length == 2 && TX_CONTROL_TYPE.equals(Type.getType(values[0]))) {
+                  txCtrl = TransactionControl.valueOf(values[1]);
```

Comment:
@chtyim I am not sure whether this is the most elegant way to get a value from an annotation using ASM. If you have suggestions to improve, please do.